training/src/main/java/wolox/training/models/Book.java

```diff
@@ -69,6 +72,7 @@ public String getGenre() {
     }
 
     public void setGenre(String genre) {
+        checkNotNull(genre, NOT_NULL_MESSAGE);
```

Comment:
Your changes are ok for me, but it would be good to check non-emptyness of the arguments.

Doing something like: "this.genre = Preconditions.checkArguments(!String.isNullOrEmpty(genre), message);"

Adapt others as needed

Reply:
Ohh okay! I'm going to improve that. Thanks you both 😄

Reply:
Changes made 🐺